### ISSUE TITLE
Support string targets for binary and multiclass problems

### ISSUE THREAD

**Issue by kmax12** [alteryx/evalml#215]
Currently if a problem has strings for the labels, evalml doesn't support it.

The problem can be reproduced with the code below

```python
import evalml
import pandas as pd

from sklearn.datasets import load_breast_cancer
data = load_breast_cancer()
X = pd.DataFrame(data["data"], columns=data["feature_names"])
y = pd.Series(data["target"])

# if we comment out the line below it works
y = y.map(lambda x: data["target_names"][x])

clf = evalml.AutoClassifier(objective="f1",
                            max_pipelines=5)

clf.fit(X, y)
```
